tests/components/imeon_inverter/test_config_flow.py

```diff
@@ -203,3 +206,30 @@ async def test_ssdp_abort(hass: HomeAssistant) -> None:
 
     assert result["type"] is FlowResultType.ABORT
     assert result["reason"] == "cannot_connect"
+
+
+@pytest.mark.parametrize(
+    ("error", "expected"),
+    [
+        (TimeoutError("Timeout"), TimeoutError),
+        (ClientError("Client error"), ClientError),
+        (ValueError("Value error"), ValueError),
+    ],
+)
+async def test_coordinator_update_method_directly(
+    hass: HomeAssistant,
+    mock_imeon_inverter: MagicMock,
+    mock_config_entry: MockConfigEntry,
+    error: Exception,
+    expected: type[Exception],
+) -> None:
+    """Test direct de la méthode d'update du coordinator."""
+    coordinator = InverterCoordinator(hass=hass, entry=mock_config_entry)
+
+    mock_imeon_inverter.update.side_effect = error
+
+    with pytest.raises(UpdateFailed) as exc_info:
+        await coordinator._async_update_data()
+
+    assert isinstance(exc_info.value.__cause__, expected)
+    mock_imeon_inverter.update.reset_mock()
```

Comment:
1. Not sure why this is a test in `test_config_flow.py`
2. We should not touch internals or create the objects itself
3. Instead create a test in test_sensor.py where we setup the intergation, and then change the `update` side effect to raise an exception, and skip time using the freezer. After that, you can check an entity's state in the state machine and expect it will be `STATE_UNAVAILABLE`